crates/forge_analyzer/src/interp.rs

```diff
@@ -676,9 +564,154 @@ impl<'cx, C: Runner<'cx>> Interp<'cx, C> {
         value: Value,
         projections: ProjectionVec,
     ) {
+        let (varid, projections) = self.get_farthest_obj(defid_block, varid, projections);
         self.value_manager
             .insert_var_with_projection(defid_block, varid, projections, value);
     }
+
+    // this function takes in an operand checks for previous values and returns a value optional
+    #[inline]
+    pub fn add_value_to_definition(&mut self, defid_block: DefId, lval: Variable, rvalue: Rvalue) {
+        if let Variable {
+            base: Base::Var(varid),
+            projections,
+        } = lval
+        {
+            let (varid, projections) = self.get_farthest_obj(defid_block, varid, projections);
+            let rval_value = self.value_from_rval(defid_block, rvalue);
+            if let Some(existing_lval) = self
+                .get_value(defid_block, varid, Some(projections.clone()))
+                .cloned()
+            {
+                // if there is an existing value...
+                match (existing_lval, rval_value) {
```

Comment:
Don't forget to delete all subprojections when an object gets overwritten, otherwise the following test case will trigger:

```javascript
let obj = { headers: { authorization: 'secret' } };
obj = {};
fetch('test.com', obj);
```

Reply:
fixed. Thank you for the help! BTreeMap worked amazing!